src/server/active_tran_server.cpp

```diff
@@ -189,7 +211,23 @@ active_tran_server::connection_handler::on_connecting ()
 
   m_prior_sender_sink_hook_func = std::bind (&active_tran_server::connection_handler::prior_sender_sink_hook, this,
 				  std::placeholders::_1);
-  log_Gl.m_prior_sender.add_sink (m_prior_sender_sink_hook_func);
+
+  auto unsent_lsa = log_Gl.m_prior_sender.add_sink (m_prior_sender_sink_hook_func);
```

Comment:
I am struggling to understand the actual value of this.
So please bear with me if this sounds silly.

To my eyes it looks like this value is destined to tell the booting up PS if it needs to do catch-up or not:
- for cold start the value of `unsent_lsa` will be exactly that from `log_Gl.prior_info.prior_lsa` which is, in turn, received from the chosen leader PS
- for PS hot-plug-ins (hot-restarts) the value will be "some value" which will be advanced enough that the booting-up PS will decide for itself that it needs to do catch-up

Are these assumptions correct?
If they are, I don't think that we need to store this `m_unsent_lsa` in the log prior sender and have all this mechanics of keeping it up to date. We can just pull the value from `log_Gl.prior_info.prior_lsa`.

Reply:
> Must we avoid this situation? No, I don't think so.

I understand your reasoning.
But I still remain to my idea that we might make it work without this. It's also possible that I miss something from your design.

Let's move forward and we'll correct/refactor later when we have a clearer picture of the mechanism - if needed. :)